### ISSUE TITLE
Fix `custom-property-pattern` false negatives for `@property` preludes

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#8430]
### What minimal example or steps are needed to reproduce the bug?

```css
@property --Foo {
  syntax: "string";
}

a {
  --Bar: a;
}
```

Ref: https://github.com/stylelint/stylelint/issues/7732#issuecomment-2548757586

### What minimal configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-standard"
  ]
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAHAThFM0BcCeABALREBiEEBwAOgHbU4DOetOAhgB7wHUiM5oAlrQDmvANx0AvnTpsqsnCQBCbNNzaTaUkABoQAM0EAbGADk2AWziIYHKylMA6MI0Z7wEWkZEIQNWgIeEDscGFooRl5uAG06IKDefjxTY2ElSG9BESJ+Ngi1KF54ggBdaQ9Mn3I0SzYcPwArRi8PWBR3RADEvnxU9Ojg43qYfl5dEqS+mDTWIirs3PYCtCKEIZGxkAqpIA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest"
}
```

### What did you expect to happen?

Both custom property names to be flagged.

### What actually happened?

Only the one inside of the ruleset is flagged:

```console
6:2-7 error Expected custom property name "--Bar" to be kebab-case (custom-property-pattern)
```

### Do you have a proposal to fix the bug?

Also check `@property` preludes.
